taichi/transforms/ir_printer.cpp

```diff
@@ -119,8 +119,9 @@ class IRPrinter : public IRVisitor {
   }
 
   void visit(FrontendAllocaStmt *alloca) override {
-    print("{}${} = alloca {}", alloca->type_hint(), alloca->id,
-          alloca->ident.name());
+    std::string shared_suffix = (alloca->is_shared) ? " shared" : "";
+    print("{}${} = alloca {}{}", alloca->type_hint(), alloca->id,
```

Comment:
Maybe `... = alloca(shared) ...`  or `... = alloca_shared ...` will be better than simply appending `shared` at the end?